### PR TITLE
Small QOL improvement for (bluespace) light replacers

### LINKED ISSUE #23540: Bluespace light replacer doens't replace from far if there is a object in the same tile as the light fixture.
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1619

### Issue Description

When trying to replace a light from far with the bluespace light replacer, even if has a big object in the same tiles as the light fixture, the replacement don't work because you need to click not like on the light fixture, but in the middle that the light "need" to be.

image

### What did you expect to happen?

Replace the light

### What happened instead?

It not get replaced

### Why is this bad/What are the consequences?

Using the bluespace light replacer is good to not having to enter directly in some room to replace the light but this happening makes the bluespace light replacer like useless. You need to click directly in the middle for this to work and if have like any object in the middle it doens't.

### Steps to reproduce the issue.

Find like a ligt fixture with a bed on something in the same tile and the light fixture and try to replace the light with the bluespace light replacer.

### When did the problem start happening?

No ideia, found that today.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Light replacers will now also work when you click on an object on the tile, meaning you don't have to pixelhunt for the turf.

## Why It's Good For The Game
It can be very frustrating that clicking on the light fixture will not work, as it currently doesn't accept any other input than turfs.
Resolves #23540

## Testing
Could replace the broken light by clicking on the lighter that I dropped on the turf

## Changelog
:cl:
tweak: Light replacers now work on anything on a tile, instead of only working when clicking a tile
/:cl:

### CODE REVIEW

**Review by Henri215 — APPROVED** [review]
LGTM

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -231,13 +231,16 @@
 	. = ..()
 	if(!proximity && !bluespace_toggle)
 		return
-	if(!isturf(T))
+	if(!isobj(T) || !isturf(T))
 		return
+	var/turf/replace_turf = T
+	if(isobj(T))
+		replace_turf = get_turf(T)
```
Comment (Contrabang) [inline code/game/objects/items/devices/lightreplacer.dm:238]: Could make it work for mobs too
```suggestion
	var/turf/replace_turf = get_turf(T)
	if(!istype(T))
		return
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/lightreplacer.dm (hunk 1 above)
Comment (DGamerL) [reply code/game/objects/items/devices/lightreplacer.dm:238]: Sounds good

File: code/game/objects/items/devices/lightreplacer.dm [hunk 2]
```diff
@@ -231,13 +231,14 @@
 	. = ..()
 	if(!proximity && !bluespace_toggle)
 		return
-	if(!isturf(T))
+	var/turf/replace_turf = get_turf(T)
+	if(!istype(T))
```
Comment (Contrabang) [inline code/game/objects/items/devices/lightreplacer.dm:235]:
```suggestion
	if(!istype(replace_turf))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
i missed this

File: code/game/objects/items/devices/lightreplacer.dm (hunk 2 above)
Comment (DGamerL) [reply code/game/objects/items/devices/lightreplacer.dm:235]: I did aswell, whoops

**Review by Contrabang — APPROVED** [review]

File: code/game/objects/items/devices/lightreplacer.dm [hunk 3]
```diff
@@ -231,13 +231,14 @@
 	. = ..()
 	if(!proximity && !bluespace_toggle)
 		return
-	if(!isturf(T))
+	var/turf/replace_turf = get_turf(T)
```
Comment (lewcc) [inline code/game/objects/items/devices/lightreplacer.dm:234]: Can you rename the variable from `T` to something more descriptive here?

**Review by lewcc — APPROVED** [review]

**Pull request merged**